### DISCUSSION TITLE
overriding equals() method in java

### DISCUSSION THREAD

**Discussion by kateryna216** [community/community#21390] [Other Feature Feedback, Questions, & Ideas]
<p>public boolean equals(Object otherObject) { super.equals(otherObject); boolean isEqual = false; if(otherObject != null &amp;&amp; otherObject instanceof Puppy) { Puppy otherPuppy = (Puppy) otherObject; if((this.dep.equals(otherPuppy.dep)) &amp;&amp; (this.getName().equals(otherPuppy.getName()))) { isEqual = true; } } return isEqual; }</p>
<p>Puppy class extends Dog class. Dog class extends Animal class. In Animal class method getName() was declared. I have error (The method getName() is undefined for the type Puppy). Why</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/f/fef6e46d6ff0151187ca2989857e280bcf479efc.png" data-download-href="/uploads/short-url/AnweZsP7eg0hqWmFoY9VqHsKQWM.png?dl=1" title="Screen Shot 2019-03-10 at 6.35.03 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/48331356/181063985-4ecaa575-06d8-4717-8f7c-dac0d97bec63.png" alt="Screen Shot 2019-03-10 at 6.35.03 PM.png" data-base62-sha1="AnweZsP7eg0hqWmFoY9VqHsKQWM" width="690" height="379" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/f/fef6e46d6ff0151187ca2989857e280bcf479efc_2_690x379.png, https://aws1.discourse-cdn.com/github/optimized/2X/f/fef6e46d6ff0151187ca2989857e280bcf479efc_2_1035x568.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/f/fef6e46d6ff0151187ca2989857e280bcf479efc_2_1380x758.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/f/fef6e46d6ff0151187ca2989857e280bcf479efc_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-03-10 at 6.35.03 PM.png</span><span class="informations">1876×1032 127 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by mpboom** [comment]
<p>Hi <a class="mention" href="/kateryna216">@kateryna216</a> ,</p>
<p>There simply  <strong>is</strong> not method called **get_name() **for your  <strong>Puppy</strong> class. </p>

**Comment by Wabri** [comment, marked as answer]
<p>Hi <a class="mention" href="/kateryna216">@kateryna216</a>,</p>
<p>Welcome to the Github Community Forum!</p>
<p>The getName() method is declared in the parent Animal class, so with this in mind what access modifiers did you assign to getName()? If is a private method you can use that only in a Animal object. To use only in a ereditary class you can assign a protected modifiers. To use everywhere assign the public modifier.</p>
<p><strong>Private method</strong></p>
<pre><code>private String getName() {
            return "something";
}
</code></pre>
<p><strong>Protected method</strong></p>
<pre><code>protected String getName() {
            return "something";
}
</code></pre>
<p><strong>Public method</strong></p>
<pre><code>public String getName() {
            return "something";
}
</code></pre>
<p>Hope to solve your problem :slight_smile:</p>
<p>-Gabriele-</p>
<blockquote>
<p>Mark helpful posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great content!</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>

**Comment by mpboom** [comment]
<p>Great explaination <a class="mention" href="/wabri">@wabri</a> !</p>
